### PR TITLE
vyos_command 2.3

### PR DESCRIPTION
**Pull request by Qalthos**
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
vyos_command

##### ANSIBLE VERSION

```
2.3
```

##### SUMMARY
Refactor vyos_command to persistent connections

### CODE REVIEW

**Comment by gundalow** [comment]
I recently found out there is a better view of failing tests in shippable

https://app.shippable.com/runs/5849bea0c42d210f009d89d6/2/tests

File: lib/ansible/modules/network/vyos/vyos_system.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+  returned: always
+  type: str
+  sample: "0:00:10.469466"
+"""
+
+EXAMPLES = """
+- name: configure hostname and domain-name
+  vyos_system:
+    hostname: vyos01
+    domain_name: eng.ansible.com
```
Comment (gundalow) [inline lib/ansible/modules/network/vyos/vyos_system.py]: foo.example.com

File: lib/ansible/modules/network/vyos/vyos_system.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+"""
+
+RETURN = """
+commands:
+  description: The list of configuration mode commands to send to the device
+  returned: always
+  type: list
+  sample:
+    - set system hostname vyos01
+    - set system domain-name eng.ansible.com
```
Comment (gundalow) [inline lib/ansible/modules/network/vyos/vyos_system.py]: foo.example.com

File: lib/ansible/modules/network/vyos/vyos_config.py [hunk 3]
```diff
@@ -217,36 +204,19 @@ def sanitize_config(config, result):
...
+    config = get_config(module)
 
     # create the candidate config object from the arguments
     candidate = get_candidate(module)
 
     # create loadable config that includes only the configuration updates
     updates = diff_config(candidate, config)
 
-    result['updates'] = updates
+    result['commands'] = updates
```
Comment (gundalow) [inline lib/ansible/modules/network/vyos/vyos_config.py]: @privateip did we make a decision on returning `updates` vs `commands`?

File: lib/ansible/module_utils/vyos.py [hunk 4]
```diff
@@ -26,87 +26,63 @@
...
-    CLI_ERRORS_RE = [
-        re.compile(r"\n\s*Invalid command:"),
-        re.compile(r"\nCommit failed"),
-        re.compile(r"\n\s+Set failed"),
-    ]
+def run_commands(module, commands, check_rc=True):
+    assert isinstance(commands, list), 'commands must be a list'
+    responses = list()
 
-    TERMINAL_LENGTH = os.getenv('ANSIBLE_VYOS_TERMINAL_LENGTH', 10000)
```
Comment (gundalow) [inline lib/ansible/module_utils/vyos.py]: Where has this moved to?

**Review by gundalow — COMMENTED** [review]
I know this is WIP, the the bot pinged me about it so I thought I'd cast my eye over it

File: lib/ansible/module_utils/vyos.py (hunk 4 above)
Comment (Qalthos) [reply lib/ansible/module_utils/vyos.py]: evidently plugins/terminal/vyos.py, will restore there.

File: lib/ansible/module_utils/vyos2.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
+# This code is part of Ansible, but is an independent component.
+# This particular file snippet, and this file snippet only, is BSD licensed.
+# Modules you write using this snippet, which is embedded dynamically by Ansible
+# still belong to the author of the module, and may assign their own license
+# to the complete work.
+#
+# Copyright (c) 2015 Peter Sprygada, <psprygada@ansible.com>
```
Comment (gundalow) [inline lib/ansible/module_utils/vyos2.py]: Should we put a Red Hat copyright notice here

File: lib/ansible/modules/network/vyos/vyos_command.py [hunk 6]
```diff
@@ -90,19 +92,9 @@
...
-# Note: examples below use the following provider dict to handle
-#       transport and authentication to the node.
-vars:
-  cli:
-    host: "{{ inventory_hostname }}"
-    username: vyos
-    password: vyos
-    transport: cli
-
 - vyos_command:
```
Comment (gundalow) [inline lib/ansible/modules/network/vyos/vyos_command.py]: For examples we are standardising on 

```
- name: describe
  module:
    arg1:
    arg2
```

File: lib/ansible/plugins/terminal/vyos.py [hunk 7]
```diff
@@ -38,9 +38,12 @@ class TerminalModule(TerminalBase):
         re.compile(r"\n\s+Set failed"),
     ]
 
+    terminal_length = os.getenv('ANSIBLE_VYOS_TERMINAL_LENGTH', 10000)
```
Comment (gundalow) [inline lib/ansible/plugins/terminal/vyos.py:42]: Thinking about this a bit more, should this be moved to `lib/ansible/constants.py`

**Review by gundalow — CHANGES_REQUESTED** [review]
needs_revision

File: lib/ansible/plugins/terminal/vyos.py (hunk 7 above)
Comment (gundalow) [reply lib/ansible/plugins/terminal/vyos.py:42]: It's correct where it is

**Review by gundalow — APPROVED** [review]
shipit

**Pull request merged**